### ISSUE TITLE
Fog::Service::NotFound openstack has no storage service

### ISSUE THREAD

**Issue by leemour** [carrierwaveuploader/carrierwave#2340]
I'm using `fog-openstack` and it fails to load `storage` service as it should require `fog/openstack/storage/storage` to work but `fog/openstack/storage` is required. I have  the following configuration:
```
CarrierWave.configure do |config|
  if Rails.env.test? || Rails.env.cucumber?
    config.storage = :file
    config.enable_processing = false
  else
    config.asset_host   = Rails.application.secrets.asset_host
    config.fog_provider = 'fog/openstack'
    config.fog_credentials = {
      provider:           'OpenStack',
      openstack_auth_url: 'https://api.selcdn.ru/v3',
      openstack_username: Rails.application.secrets.selectel[:username],
      openstack_api_key:  Rails.application.secrets.selectel[:password],
    }
    config.fog_directory  = Rails.application.secrets.selectel_container
    config.fog_public     = true
    config.fog_attributes = {
      cache_control: "public, max-age=#{365.days.to_i}"
    }
  end
end
```
And I get this error when starting Rails:

```
[fog][DEPRECATION] Unable to require fog/openstack/storage
[fog][DEPRECATION] The format fog/storage/openstack is deprecated
Exiting
/home/leemour/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fog-core-2.1.2/lib/fog/core/services_mixin.rb:18:in `rescue in new': openstack has no storage service (Fog::Service::NotFound)
	from /home/leemour/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/fog-core-2.1.2/lib/fog/core/services_mixin.rb:8:in `new'
	from /home/leemour/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/carrierwave-1.2.3/lib/carrierwave/uploader/configuration.rb:126:in `eager_load_fog'
	from /home/leemour/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/carrierwave-1.2.3/lib/carrierwave/uploader/configuration.rb:139:in `fog_credentials='
	from /home/leemour/Ruby/burobiz/config/initializers/carrierwave.rb:8:in `block in <main>'
	from /home/leemour/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/carrierwave-1.2.3/lib/carrierwave/uploader/configuration.rb:161:in `configure'
	from /home/leemour/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/carrierwave-1.2.3/lib/carrierwave.rb:14:in `configure'
	from /home/leemour/Ruby/burobiz/config/initializers/carrierwave.rb:1:in `<main>'
```
I wonder if it's a problem with Carrierwave of with Fog? After overriding `Fog::ServicesMixin#require_service_provider_library` (to do nothing) it works as expected. I'm surprised not find an existing issue, maybe I'm doing smth wrong?

**Comment by VincentSim** [comment]
I have the same bug 👎

**Comment by VincentSim** [comment]
@leemour how did you handle it ?

**Comment by leemour** [comment]
@VincentSim as I said, I've just overridden `Fog::ServicesMixin#require_service_provider_library` (to do nothing).

**Comment by dmeremyanin** [comment]
This seems to be fixed in fog-openstack 1.0.8
